File: dotnet/src/EventGraph/CodeGraph/CodeGraphEventTypes.cs
namespace EventGraph.CodeGraph;

/// <summary>All 35 Code Graph event type constants. All use the "codegraph." prefix.</summary>
public static class CodeGraphEventTypes
{
    // ── Data events ──────────────────────────────────────────────────────
    public static readonly EventType EntityCreated = new("codegraph.entity.created");
    public static readonly EventType EntityUpdated = new("codegraph.entity.updated");
    public static readonly EventType EntityDeleted = new("codegraph.entity.deleted");
    public static readonly EventType PropertySet = new("codegraph.entity.property.set");
    public static readonly EventType RelationCreated = new("codegraph.entity.relation.created");
    public static readonly EventType StateChanged = new("codegraph.state.changed");

    // ── Logic events ─────────────────────────────────────────────────────
    public static readonly EventType TransformApplied = new("codegraph.logic.transform.applied");
    public static readonly EventType ConditionEvaluated = new("codegraph.logic.condition.evaluated");
    public static readonly EventType SequenceStarted = new("codegraph.logic.sequence.started");
    public static readonly EventType SequenceCompleted = new("codegraph.logic.sequence.completed");
    public static readonly EventType LoopIterated = new("codegraph.logic.loop.iterated");
    public static readonly EventType TriggerFired = new("codegraph.logic.trigger.fired");
    public static readonly EventType ConstraintViolated = new("codegraph.logic.constraint.violated");

    // ── IO events ────────────────────────────────────────────────────────
    public static readonly EventType QueryExecuted = new("codegraph.io.query.executed");
    public static readonly EventType CommandExecuted = new("codegraph.io.command.executed");
    public static readonly EventType SubscriptionCreated = new("codegraph.io.subscribe.created");
    public static readonly EventType AuthorizeChecked = new("codegraph.io.authorize.checked");
    public static readonly EventType SearchExecuted = new("codegraph.io.search.executed");
    public static readonly EventType InteropCalled = new("codegraph.io.interop.called");

    // ── UI events ────────────────────────────────────────────────────────
    public static readonly EventType ViewRendered = new("codegraph.ui.view.rendered");
    public static readonly EventType ActionTriggered = new("codegraph.ui.action.triggered");
    public static readonly EventType NavigationChanged = new("codegraph.ui.navigation.changed");
    public static readonly EventType FeedbackShown = new("codegraph.ui.feedback.shown");
    public static readonly EventType AlertRaised = new("codegraph.ui.alert.raised");
    public static readonly EventType ConfirmationRequested = new("codegraph.ui.confirmation.requested");

    // ── Aesthetic events ─────────────────────────────────────────────────
    public static readonly EventType ThemeApplied = new("codegraph.aesthetic.theme.applied");

    // ── Temporal events ──────────────────────────────────────────────────
    public static readonly EventType UndoRequested = new("codegraph.temporal.undo.requested");
    public static readonly EventType RetryAttempted = new("codegraph.temporal.retry.attempted");

    // ── Resilience events ────────────────────────────────────────────────
    public static readonly EventType FallbackActivated = new("codegraph.resilience.fallback.activated");
    public static readonly EventType OfflineSynced = new("codegraph.resilience.offline.synced");

    // ── Structural events ────────────────────────────────────────────────
    public static readonly EventType ScopeEntered = new("codegraph.structural.scope.entered");
    public static readonly EventType FormatApplied = new("codegraph.structural.format.applied");

    // ── Social events ────────────────────────────────────────────────────
    public static readonly EventType PresenceChanged = new("codegraph.social.presence.changed");
    public static readonly EventType SalienceScored = new("codegraph.social.salience.scored");
    public static readonly EventType ConsequencePreviewed = new("codegraph.ui.confirmation.previewed");

    /// <summary>Returns all 35 registered Code Graph event types.</summary>
    public static List<EventType> AllCodeGraphEventTypes() => new()
    {
        // Data
        EntityCreated, EntityUpdated, EntityDeleted,
        PropertySet, RelationCreated, StateChanged,
        // Logic
        TransformApplied, ConditionEvaluated,
        SequenceStarted, SequenceCompleted,
        LoopIterated, TriggerFired, ConstraintViolated,
        // IO
        QueryExecuted, CommandExecuted,
        SubscriptionCreated, AuthorizeChecked,
        SearchExecuted, InteropCalled,
        // UI
        ViewRendered, ActionTriggered, NavigationChanged,
        FeedbackShown, AlertRaised, ConfirmationRequested,
        // Aesthetic
        ThemeApplied,
        // Temporal
        UndoRequested, RetryAttempted,
        // Resilience
        FallbackActivated, OfflineSynced,
        // Structural
        ScopeEntered, FormatApplied,
        // Social
        PresenceChanged, SalienceScored, ConsequencePreviewed,
    };
}

File: go/pkg/event/codegraph_event_types.go
package event

import "github.com/transpara-ai/eventgraph/go/pkg/types"

// Code graph event type constants. All use the "codegraph." prefix.
var (
	// Data events
	EventTypeCodeGraphEntityDefined  = types.MustEventType("codegraph.entity.defined")
	EventTypeCodeGraphEntityModified = types.MustEventType("codegraph.entity.modified")
	EventTypeCodeGraphEntityRelated  = types.MustEventType("codegraph.entity.related")
	EventTypeCodeGraphStateTransitioned = types.MustEventType("codegraph.state.transitioned")

	// Logic events
	EventTypeCodeGraphTransformApplied     = types.MustEventType("codegraph.logic.transform.applied")
	EventTypeCodeGraphConditionEvaluated   = types.MustEventType("codegraph.logic.condition.evaluated")
	EventTypeCodeGraphSequenceExecuted     = types.MustEventType("codegraph.logic.sequence.executed")
	EventTypeCodeGraphLoopIterated         = types.MustEventType("codegraph.logic.loop.iterated")
	EventTypeCodeGraphTriggerFired         = types.MustEventType("codegraph.logic.trigger.fired")
	EventTypeCodeGraphConstraintViolated   = types.MustEventType("codegraph.logic.constraint.violated")

	// IO events
	EventTypeCodeGraphQueryExecuted       = types.MustEventType("codegraph.io.query.executed")
	EventTypeCodeGraphCommandExecuted     = types.MustEventType("codegraph.io.command.executed")
	EventTypeCodeGraphCommandRejected     = types.MustEventType("codegraph.io.command.rejected")
	EventTypeCodeGraphSubscribeRegistered = types.MustEventType("codegraph.io.subscribe.registered")
	EventTypeCodeGraphAuthorizeGranted    = types.MustEventType("codegraph.io.authorize.granted")
	EventTypeCodeGraphAuthorizeDenied     = types.MustEventType("codegraph.io.authorize.denied")
	EventTypeCodeGraphSearchExecuted      = types.MustEventType("codegraph.io.search.executed")
	EventTypeCodeGraphInteropSent         = types.MustEventType("codegraph.io.interop.sent")
	EventTypeCodeGraphInteropReceived     = types.MustEventType("codegraph.io.interop.received")

	// UI events
	EventTypeCodeGraphViewRendered        = types.MustEventType("codegraph.ui.view.rendered")
	EventTypeCodeGraphActionInvoked       = types.MustEventType("codegraph.ui.action.invoked")
	EventTypeCodeGraphNavigationTriggered = types.MustEventType("codegraph.ui.navigation.triggered")
	EventTypeCodeGraphFeedbackEmitted     = types.MustEventType("codegraph.ui.feedback.emitted")
	EventTypeCodeGraphAlertDispatched     = types.MustEventType("codegraph.ui.alert.dispatched")
	EventTypeCodeGraphDragCompleted       = types.MustEventType("codegraph.ui.drag.completed")
	EventTypeCodeGraphSelectionChanged    = types.MustEventType("codegraph.ui.selection.changed")
	EventTypeCodeGraphConfirmationResolved = types.MustEventType("codegraph.ui.confirmation.resolved")

	// Aesthetic events
	EventTypeCodeGraphSkinApplied = types.MustEventType("codegraph.aesthetic.skin.applied")

	// Temporal events
	EventTypeCodeGraphUndoExecuted    = types.MustEventType("codegraph.temporal.undo.executed")
	EventTypeCodeGraphRetryAttempted  = types.MustEventType("codegraph.temporal.retry.attempted")

	// Resilience events
	EventTypeCodeGraphOfflineEntered = types.MustEventType("codegraph.resilience.offline.entered")
	EventTypeCodeGraphOfflineSynced  = types.MustEventType("codegraph.resilience.offline.synced")

	// Structural events
	EventTypeCodeGraphScopeDefined = types.MustEventType("codegraph.structural.scope.defined")

	// Social events
	EventTypeCodeGraphPresenceUpdated    = types.MustEventType("codegraph.social.presence.updated")
	EventTypeCodeGraphSalienceTriggered  = types.MustEventType("codegraph.social.salience.triggered")
)

// AllCodeGraphEventTypes returns all registered code graph event types.
func AllCodeGraphEventTypes() []types.EventType {
	return []types.EventType{
		// Data
		EventTypeCodeGraphEntityDefined, EventTypeCodeGraphEntityModified,
		EventTypeCodeGraphEntityRelated, EventTypeCodeGraphStateTransitioned,
		// Logic
		EventTypeCodeGraphTransformApplied, EventTypeCodeGraphConditionEvaluated,
		EventTypeCodeGraphSequenceExecuted, EventTypeCodeGraphLoopIterated,
		EventTypeCodeGraphTriggerFired, EventTypeCodeGraphConstraintViolated,
		// IO
		EventTypeCodeGraphQueryExecuted, EventTypeCodeGraphCommandExecuted,
		EventTypeCodeGraphCommandRejected, EventTypeCodeGraphSubscribeRegistered,
		EventTypeCodeGraphAuthorizeGranted, EventTypeCodeGraphAuthorizeDenied,
		EventTypeCodeGraphSearchExecuted, EventTypeCodeGraphInteropSent,
		EventTypeCodeGraphInteropReceived,
		// UI
		EventTypeCodeGraphViewRendered, EventTypeCodeGraphActionInvoked,
		EventTypeCodeGraphNavigationTriggered, EventTypeCodeGraphFeedbackEmitted,
		EventTypeCodeGraphAlertDispatched, EventTypeCodeGraphDragCompleted,
		EventTypeCodeGraphSelectionChanged, EventTypeCodeGraphConfirmationResolved,
		// Aesthetic
		EventTypeCodeGraphSkinApplied,
		// Temporal
		EventTypeCodeGraphUndoExecuted, EventTypeCodeGraphRetryAttempted,
		// Resilience
		EventTypeCodeGraphOfflineEntered, EventTypeCodeGraphOfflineSynced,
		// Structural
		EventTypeCodeGraphScopeDefined,
		// Social
		EventTypeCodeGraphPresenceUpdated, EventTypeCodeGraphSalienceTriggered,
	}
}
